### PR TITLE
made concurent task fail 

### PR DESCRIPTION
**Pull request by Trial97**
this is not really tested for all cases, but it should be ok I think

The concurrent task never fails even if one or all tasks fail. This PR fixes that but it may also introduce some bugs as in most cases we check for if the tasks failed but we do not expect to fail.

### CODE REVIEW

**Review by getchoo — APPROVED** [review]

**Comment by Ryex** [comment]
before we merge this we need to run through places concurrent tasks can fail and be sure that failure is handled correctly

**Comment by Trial97** [comment]
With this, all failed signals that were related to ConcurrentTask/NetJob were connected(from my search through code).
Obviously, I tried to warn the user of the fail the best I could. 
If a signal needs to be changed or added, please notify me and I will update this.
As Ryex said this needs to be tested before merging, as this small change brings a bunch of changes to the tasks subsystem.
Maybe rethink the milestone assignment as it may push the 7.2 to a later date.

File: launcher/tasks/ConcurrentTask.cpp [hunk 1]
```diff
@@ -130,6 +134,9 @@ void ConcurrentTask::startNext()
 
     connect(next.get(), &Task::succeeded, this, [this, next]() { subTaskSucceeded(next); });
     connect(next.get(), &Task::failed, this, [this, next](QString msg) { subTaskFailed(next, msg); });
+    // this should never happen but if it does better to fail the task that being stuck
+    // most
```
Comment (getchoo) [inline launcher/tasks/ConcurrentTask.cpp]:
```suggestion
    // this should never happen but if it does, it's better to fail the task than get stuck
```

**Review by getchoo — COMMENTED** [review]
i'll get to testing this soon :)

**Comment by Trial97** [comment]
Closing this in favor of https://github.com/PrismLauncher/PrismLauncher/pull/1694 (at least I fully tested that one)

**Pull request closed without merging**